File: glider/ros/node.cpp
/*
*
*
*/

#include "glider/ros/factor_node.hpp"
#include <rclcpp_components/register_node_macro.hpp>
#include <rclcpp/rclcpp.hpp>


int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    rclcpp::NodeOptions options;

    try
    {
        rclcpp::Node = std::make_shared<glider::FactorNode> node(options);
        rclcpp::executors::MultiThreadedExecutor executor;

        executor.add_node(node);

        executor.spin()

        rclcpp::shutdown();
    }
    catch (const std::exception& e)
    {
        std::cerr << "[GLIDER] Caught error " << e.what() << std::endl;
    }
    return 0;
}

File: glider-viz/glider_viz/node.py
"""
    Jason Hughes
    September 2025

    launch the node
"""

import rclpy
from rclpy.executors import SingleThreadedExecutor
from glider_viz.map_node import MapNode


def main(args=None) -> None:
    rclpy.init(args=args)

    node = MapNode()
    executor = SingleThreadedExecutor()

    executor.add_node(node)

    try:
        executor.spin()
    finally:
        executor.shutdown()
        node.destroy_node()
        rclpy.shutdown()

if __name__ == "__main__":
    main()
